src/api/Metadata.js

```diff
@@ -137,14 +136,15 @@ class Metadata extends File {
      * @param {string} scope - metadata scope
      * @return {Object} array of metadata templates
      */
-    async getTemplates(id: string, scope: string): Promise<Array<MetadataEditorTemplate>> {
+    async getTemplates(id: string, scope: string, isInstanceId?: boolean): Promise<Array<MetadataEditorTemplate>> {
+        this.errorCode = ERROR_CODE_FETCH_METADATA_TEMPLATES;
         let templates = {};
         try {
             templates = await this.xhr.get({
-                url: this.getMetadataTemplateUrl(scope),
+                url: this.getMetadataTemplateUrl(scope, isInstanceId),
                 id: getTypedFileId(id),
                 params: {
-                    limit: 1000,
+                    limit: 1000, // internal hard limit is 500
```

Comment:
Can we slap this value into a meaningful constant? I'd even argue in a config/constants file that an external dev can make quick modifications to, to allow for different limits.